### DISCUSSION TITLE
Support for pre-compressed assets and brotli compression

### DISCUSSION THREAD

**Discussion by curbengh** [community/community#21655] [Other Feature Feedback, Questions, & Ideas]
<p>GitHub Pages currently support on-the-fly gzip compression. Since it’s resource-intensive, usually the compression level is not set to high/max. Users can instead opt for pre-compressing the assets using the maximum compression level. There are Jekyll plugins (<a href="https://github.com/philnash/jekyll-gzip" rel="nofollow noopener">1</a>, <a href="https://github.com/philnash/jekyll-zopfli" rel="nofollow noopener">2</a>, <a href="https://github.com/philnash/jekyll-brotli" rel="nofollow noopener">3</a> <span class="mention">@philnash</span>) that enable users to do that easily. I’m not sure Pages actually utilize pre-compressed assets or not, but it would benefit Github (in reducing CPU resource and bandwidth) if that’s the case.</p>
<p>All major browsers <a href="https://en.wikipedia.org/wiki/Brotli" rel="nofollow noopener">support</a> brotli since the end of 2017.</p>
<p>TL;DR : Suggest to support gzip and brotli pre-compressed assets and brotli on-the-fly compression.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>

**Comment by vchirikov** [comment]
<p>Was this implemented?</p>

**Reply by IhfazAlam** [reply]
Hello

**Reply by IhfazAlam** [reply]
Plz follow me

**Reply by IhfazAlam** [reply]
I'm in class 4

**Comment by curbengh** [comment]
<p>I have a <a href="https://curbengh.github.io/hexo-testing/" rel="nofollow noopener">test page</a> with pre-<a href="https://github.com/curbengh/hexo-testing/tree/gh-pages" rel="nofollow noopener">br-compressed</a> files and gh page still serves gzip.</p>

**Reply by iSeiryu** [reply]
It returns `br` now. How did you achieve it? I see your repo has both `gzip` and `br` files and browser sends `Accept-Encoding | gzip, deflate, br`.

**Reply by Semro** [reply]
@iSeiryu it fetches `br` compressed files from CDN.

**Comment by evangoad** [comment]
<p>I have done a similar test as <a class="mention" href="/curbengh">@curbengh</a>.  Still looks like Github Pages is not serving pre-compressed brotli assets.  Interestingly, you can serve pre-compressed brotli assets from Gitlab Pages, I have a demo here: <a href="http://evangoad.gitlab.io" rel="noopener nofollow ugc">evangoad.gitlab.io</a></p>

**Comment by runspired** [comment]
Any update?

**Reply by wfd2018** [reply]
I don't think that anything is happening with this issue.  I have tried a few times to deploy here but have had no luck with this brotli website that I made, but it renders well locally with http-server from npm.  I  will try Netlify and gitlab..

**Comment by jprendes** [comment]
Posting in case anyone else may find it useful.
As a stop-gap solution I created a service worker that attempts to fetch and decompress the brotli precompressed assets.
https://github.com/jprendes/service-worker-brotli
https://www.npmjs.com/package/service-worker-brotli

**Comment by funjoker** [comment]
4years+ still nothing yet ?

**Reply by cizordj** [reply]
Exactly

**Comment by mobeicanyue** [comment]
5 years still nothing yet ?

**Reply by cizordj** [reply]
There is a solution for this: Cloudflare pages. Thank me later.

**Comment by whatjakecodes** [comment]
Curious if any support is planned for this. Would be great for embedding Unity WebGL games into a Github Pages site.

**Reply by wakeupinagear** [reply]
Came here for this. Unity WebGL is enabling brotli by default (for good reasons!) and hosting a game on pages is an extremely common use-case, especially for portfolio sites.
@whatjakecodes As a stopgap, you can switch to a different compression type in Publishing Settings

**Reply by cizordj** [reply]
Don't use pre compression on Unity games if you are intending to host them on Github pages, the webserver already compresses them in gzip and if you need brotli support then move to another service like Cloudflare pages which supports brotli compression.

**Reply by zkkzkk32312** [reply]
> Don't use pre compression on Unity games if you are intending to host them on Github pages, the webserver already compresses them in gzip and if you need brotli support then move to another service like Cloudflare pages which supports brotli compression.

so when building webGL that are intended to be hosted on Github Pages, use no compression instead of Gzip? doesn't Gzip actually requires you to send Gzip related headers?

**Reply by cizordj** [reply]
GitHub Pages will automatically gzip your files for users whose browsers support it. You can verify this by uploading your files and checking the HTTP headers.

**Reply by zkkzkk32312** [reply]
> GitHub Pages will automatically gzip your files for users whose browsers support it. You can verify this by uploading your files and checking the HTTP headers.

I was able to make a build with Gzip compression and decompression fall back checked, and host it on Github pages. Size went down from 100MB to 57MB and since github supports Gzip, all you have to do is check the decompression fallback.
